src/utils/isPlainObject.js

```diff
@@ -0,0 +1,3 @@
+export default function isPlainObject(obj) {
+  return obj ? typeof obj === 'object' && Object.getPrototypeOf(obj) === Object.prototype : false;
```

Comment:
>It is perfectly fine for someone to have a React state that is not a plain object.

I thought that broke in some recent version, didn't it? Either way they don't officially support it yet.
https://github.com/facebook/react/issues/3303